### PR TITLE
CDD-2326: Migrate audit urls to cms admin

### PR DESCRIPTION
**Pull request by phill-stanley**
# Description

This PR includes and update to the audit endpoints, moving them to the cms admin app mode and adding in a permissions check so that users can only make requests to these endpoints if they are logged in as an admin user.

Fixes #CDD-2326

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests at the right levels to prove my change is effective
- [x] I have added screenshots or screen grabs where appropriate
- [x] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

### CODE REVIEW

**Comment by phill-stanley** [comment]
Screenshot 2024-12-10 at 09 46 22
Screenshot 2024-12-10 at 09 46 40

File: tests/unit/metrics/api/views/audit/test_audit_api_timeseries.py [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+        self,
+        spy_permissions_is_authenticated: mock.MagicMock,
+    ):
+        """
+        Given the application is running with an APP_MODE of `CMS_ADMIN`
+        When `get_permissions()` is called from the API timeseries view set
+        Then rest_frameworks's `IsAuthenticated()` will be called.
+        """
+        # Given
+        config.APP_MODE = "CMS_ADMIN"
```
Comment (A-Ashiq) [inline tests/unit/metrics/api/views/audit/test_audit_api_timeseries.py:22]: You are mutating the value of this here. 
It's probably gonna be safer to monkey patch the underlying env var:
```
monkeypatch.setenv(name="APP_MODE", value="CMS_ADMIN")
```

File: tests/unit/metrics/api/views/audit/test_audit_core_headline.py [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
+        self,
+        spy_permissions_is_authenticated: mock.MagicMock,
+    ):
+        """
+        Given the application is running with an APP_MODE of `CMS_ADMIN`
+        When `get_permissions()` is called from the Core headline view set
+        Then rest_frameworks's `IsAuthenticated()` will be called.
+        """
+        # Given
+        config.APP_MODE = "CMS_ADMIN"
```
Comment (A-Ashiq) [inline tests/unit/metrics/api/views/audit/test_audit_core_headline.py:22]: Same as above

File: tests/unit/metrics/api/views/audit/test_audit_core_timeseries.py [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
+        self,
+        spy_permissions_is_authenticated: mock.MagicMock,
+    ):
+        """
+        Given the application is running with an APP_MODE of `CMS_ADMIN`
+        When `get_permission()` is called from the Core timeseries view set
+        Then rest_framework's `IsAuthenticated()` will be called.
+        """
+        # Given
+        config.APP_MODE = "CMS_ADMIN"
```
Comment (A-Ashiq) [inline tests/unit/metrics/api/views/audit/test_audit_core_timeseries.py:22]: As above

File: metrics/api/urls_construction.py [hunk 4]
```diff
@@ -240,6 +240,7 @@ def construct_urlpatterns(
                 app_mode=app_mode
             )
             constructed_url_patterns += django_admin_urlpatterns
+            constructed_url_patterns += audit_api_urlpatterns
```
Comment (A-Ashiq) [inline metrics/api/urls_construction.py:243]: Can you update the tests at `tests/unit/metrics/api/test_urls_construction.py` to demonstrate these audit endpoints being added to the correct place

File: tests/unit/metrics/api/views/audit/test_audit_api_timeseries.py (hunk 1 above)
Comment (phill-stanley) [reply tests/unit/metrics/api/views/audit/test_audit_api_timeseries.py:22]: Updated these tests to use monkeypatch with context manager.

**Review by A-Ashiq — APPROVED** [review]

**Pull request merged**